spec/requests/authentication_spec.rb

```diff
@@ -240,10 +240,17 @@
   end
 
   context "Token Based Authentication" do
-    %w(sql memory cache).each do |session_store|
-      context "when using a #{session_store} session store" do
-        before { stub_settings_merge(:server => {:session_store => session_store}) }
+    before do
+      # Keep this outside of the loop below with the `let` providing the value
+      # for the stub, otherwise it will not work as expected
+      stub_settings_merge(:server => {:session_store => session_store_value})      
+      TokenStore.token_caches.clear
+    end
 
+    after { TokenStore.token_caches.clear }
+
+    %w[sql memory cache].each do |session_store|
+      context "when using a #{session_store} session store" do
```

Comment:
don't we still have to stub the session_store once we know the type (within this block)?

`stub_settings_merge(:server => {:session_store => session_store}) }`

Reply:
I think we need to stub it each time we enter this loop